dev/DynamicDependency/PackageGraphManager.h

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if !defined(PACKAGEGRAPHMANAGER_H)
+#define PACKAGEGRAPHMANAGER_H
+
+#include <PackageGraph.h>
+
+namespace MddCore
+{
+class PackageGraphManager
+{
+public:
+    static HRESULT ResolvePackageDependency(
+        PCWSTR packageDependencyId,
+        MddAddPackageDependencyOptions options,
+        wil::unique_process_heap_string& packageFullName);
+
+    static HRESULT AddToPackageGraph(
+        _In_ PCWSTR packageDependencyId,
+        INT32 rank,
+        MddAddPackageDependencyOptions options,
+        _Out_ MDD_PACKAGEDEPENDENCY_CONTEXT* packageDependencyContext,
+        _Outptr_opt_result_maybenull_ PWSTR* packageFullName);
+
+    static void RemoveFromPackageGraph(
+        MDD_PACKAGEDEPENDENCY_CONTEXT context);
+
+    static HRESULT GetPackageDependencyForContext(
+        _In_ MDD_PACKAGEDEPENDENCY_CONTEXT context,
+        wil::unique_process_heap_string& packageDependencyId);
+
+public:
+    typedef LONG (WINAPI * GetCurrentPackageInfo2Function)(
+        const UINT32 flags,
+        PackagePathType packagePathType,
+        UINT32* bufferLength,
+        BYTE* buffer,
+        UINT32* count);
+
+    static HRESULT GetCurrentPackageInfo2(
+        const UINT32 flags,
+        PackagePathType packagePathType,
+        UINT32* bufferLength,
+        BYTE* buffer,
+        UINT32* count,
+        GetCurrentPackageInfo2Function getCurrentPackageInfo2) noexcept;
+
+private:
+    static UINT32 SerializePackageInfoToBuffer(
```

Comment:
Private static methods should just go into the implementation file directly as loose functions.